persistence/gorm/serializer.go

```diff
@@ -132,3 +133,48 @@ func (AnySerializer) Scan(ctx context.Context, field *schema.Field, dst reflect.
 	field.ReflectValueOf(ctx, dst).Set(reflect.ValueOf(&a))
 	return
 }
+
+// StructpbValueSerializer is a GORM serializer that allows the serialization and deserialization of the
+// google.protobuf.Value protobuf message type.
+type StructpbValueSerializer struct{}
+
+// Value implements https://pkg.go.dev/gorm.io/gorm/schema#SerializerValuerInterface to indicate
+// how this struct will be saved into an SQL database field.
+func (StructpbValueSerializer) Value(_ context.Context, _ *schema.Field, _ reflect.Value, fieldValue interface{}) (interface{}, error) {
+	var (
+		v  *structpb.Value
+		ok bool
+	)
+
+	if fieldValue == nil {
```

Comment:
We had some issues with "nil" in other serialisers, we need to double check if the nil checks are the same in every serializer